molgenis/molgenis-emx2
# feat: enable creation and use of auth token for programmatic access. M8846

closes #902

- [x] generalize settings to apply to schema, table, user in uniform way
- [x]  make excel writer streaming (don't know why it was needed in this PR; we could push it seperate)
- [x]  refactor settings internally from List<Setting> to Map<String,String>, removing all kinds of unnecessary code
- [x] implement token mechanism
- [x] Added user interface under user account
- [x] Check if works for non admin user

Out of scope: update user interface to grant/revoke your tokes tokens as part of your user account

How to test:
- in user interface click account (name user)
- Use graphql api createtoken
- Follow example from the test https://github.com/molgenis/molgenis-emx2/blob/95db81e219869ca806e326d142102c16fa810c14/backend/molgenis-emx2-webapi/src/test/java/org.molgenis.emx2.web/WebApiSmokeTests.java#L526
- GraphQL api settings, createToken, user{settings}, see manual
- in user interface click after sign in on the user; next to password you can create token
- for temp token in graphql {_session{token}} and then use it
